test/rdoc/test_rdoc_rubygems_hook.rb

```diff
@@ -1,40 +1,46 @@
 # frozen_string_literal: true
-require 'rubygems/test_case'
+require "rubygems"
+require "fileutils"
+require "tmpdir"
 require 'rdoc/rubygems_hook'
+require "test/unit"
 
-class TestRDocRubygemsHook < Gem::TestCase
-  unless method_defined?(:assert_path_exist)
-    alias assert_path_exist assert_path_exists
-  end
-  unless method_defined?(:assert_path_not_exist)
-    alias assert_path_not_exist refute_path_exists
-  end
-  unless method_defined?(:assert_raise)
-    alias assert_raise assert_raises
-  end
-
+class TestRDocRubygemsHook < Test::Unit::TestCase
   def setup
-    super
+    FileUtils.mkdir_p File.expand_path("tmp")
 
-    @a = util_spec 'a', 2 do |s|
+    @a = Gem::Specification.new do |s|
+      s.platform    = Gem::Platform::RUBY
+      s.name        = "a"
+      s.version     = 2
       s.rdoc_options = %w[--main MyTitle]
       s.extra_rdoc_files = %w[README]
     end
+    @tempdir = Dir.mktmpdir("test_rubygems_", File.expand_path("tmp"))
 
-    write_file File.join(@tempdir, 'lib', 'a.rb')
-    write_file File.join(@tempdir, 'README')
+    @a.instance_variable_set(:@doc_dir, File.join(@tempdir, "doc"))
+    @a.instance_variable_set(:@gem_dir, File.join(@tempdir, "a-2"))
+    @a.instance_variable_set(:@full_gem_path, File.join(@tempdir, "a-2"))
+    @a.loaded_from = File.join(@tempdir, 'a-2', 'a-2.gemspec')
 
-    install_gem @a
+    FileUtils.mkdir_p File.join(@tempdir, 'a-2', 'lib')
+    FileUtils.touch   File.join(@tempdir, 'a-2', 'lib', 'a.rb')
+    FileUtils.touch   File.join(@tempdir, 'a-2', 'README')
 
     @hook = RDoc::RubygemsHook.new @a
 
     begin
       RDoc::RubygemsHook.load_rdoc
     rescue Gem::DocumentError => e
-      skip e.message
+      pend e.message
```

Comment:
This is an interesting change, would `pend` be interchangeable with `omit` here?

Reply:
Ah, I was late to the conversation, sorry about that.

Reply:
I'm not sure what differences `pend` and `omit`.

@kou Do you have any opinion about them?

Reply:
`omit` is suitable for this case.
`omit` for omitting the test because the test isn't needed to be ran.
`pend` for omitting the test temporary. The test should be fixed later.

Reply:
Generally, `pend` is used with a block.

If the given block is failed, it's expected for `pend`. The test is marked as "pending".

If the given block is not failed, it's unexpected for `pend`. The test is marked as "failed".

```ruby
require "test-unit"

class T < Test::Unit::TestCase
  test "pend: pass" do
    pend "XXX" do
      assert_equal("AAA", "AAA")
    end
  end

  test "pend: failure" do
    pend "XXX" do
      assert_equal("AAA", "BBB")
    end
  end
end
```

Reply:
@kou Thanks! I understood them.